src/middlewared/middlewared/plugins/system.py

```diff
@@ -23,6 +24,163 @@
 SYSTEM_READY = False
 
 
+class SytemAdvancedService(ConfigService):
+
+    class Config:
+        datastore = 'system.advanced'
+        datastore_prefix = 'adv_'
+        datastore_extend = 'system.advanced.system_advanced_extend'
+        namespace = 'system.advanced'
+
+    async def serial_port_choices(self):
```

Comment:
`choices.SERIAL_CHOICES` should also be refactored to use this method.

Also this method should be public (add `@accepts()`) so it can be used by new UI.